tools/rtpsrelay/RelayHandler.cpp

```diff
@@ -249,20 +249,13 @@ int RelayHandler::handle_timeout(const ACE_Time_Value& ace_now, const void* ptr)
       if (ret != DDS::RETCODE_OK) {
         ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) %N:%l ERROR: RelayHandler::handle_timeout %C failed to write handler statistics\n"), name_.c_str()));
       }
-    }
 
-    if (config_.participant_statistics_writer()) {
-      for (auto& p : participant_statistics_) {
-        auto& stats = p.second;
-        stats.application_participant_guid(handler_statistics_._application_participant_guid);
-        stats.name(name_);
-        stats.address(addr_to_string(p.first));
-        stats._interval._sec = dds_duration.sec;
-        stats._interval._nanosec = dds_duration.nanosec;
-
-        const auto ret = config_.participant_statistics_writer()->write(stats, DDS::HANDLE_NIL);
-        if (ret != DDS::RETCODE_OK) {
-          ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) %N:%l ERROR: RelayHandler::handle_timeout %C failed to write participant statistics\n"), name_.c_str()));
+      if (config_.publish_participant_statistics()) {
+        size_t idx = 0;
```

Comment:
Where does `idx` change?  Should the range-based `for` below be a classic `for` or did I miss something?